File: Data Structures & Algorithms/sort-an-array/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    sortArray(nums) {
        if(nums.length <= 1)
            return nums;
        
        return this.mergeSort(nums);
    }

     mergeSort(nums){
        if(nums.length <= 1) 
            return nums;
        
        const mid = nums.length >> 1;
        const left = this.mergeSort(nums.slice(0, mid));
        const right = this.mergeSort(nums.slice(mid));

        return this.merge(left, right);
    }

    merge(a, b){
        let res = new Array(a.length + b.length);
        let i = 0, j = 0, k = 0;

        while(i < a.length && j < b.length)
        {
            if(a[i]  <= b[j])
                res[k++] = a[i++];
            else
                res[k++] = b[j++];
        }

        while (i < a.length){
            res[k++] = a[i++];
        }

        while (j < b.length){
            res[k++] = b[j++];
        }

        return res;
    }
}

File: Data Structures & Algorithms/remove-element/submission-2.cs
public class Solution {
    public int RemoveElement(int[] nums, int val)
    {
        var ptr1 = 0;
        var k = 0;

        for(int i = ptr1; i < nums.Length; i++)
        {
            if(nums[i] != val) 
            {
                ptr1 = i + 1;    
                k++; 
                continue;
            }

            for(var j = i + 1; j < nums.Length; j++)
            {
                if(nums[j] != val)
                {
                    (nums[i], nums[j]) = (nums[j], nums[i]);
                    k++;
                    break;
                }
            }
        }
        
        return k;
    }
}